### DISCUSSION TITLE
what is the the difference between pathname and route of router object. ?

### DISCUSSION THREAD

**Discussion by deresegetachew** [vercel/next.js#14521] [Help]
The router object returned by both `useRouter` and `withRouter`: has properties `pathname` and `route`, in the documentation, the property route is not even mentioned; I was wondering what the difference between the two is since they have the same values.

**Comment by dekisr** [comment]
Still unanswered...

Did you find something @deresegetachew ?

**Comment by ghost** [comment]
Looks like `route` is served as a component key, and `pathname` is URL path with possible trailing `/`

From the [source code](https://github.com/vercel/next.js/blob/57a87050e77e9398842a78e43d48b26542642330/packages/next/shared/lib/router/router.ts#L673)
```ts
 // represents the current component key
 this.route = removePathTrailingSlash(pathname)
```

```ts
/**
 * Removes the trailing slash of a path if there is one. Preserves the root path `/`.
 */
export function removePathTrailingSlash(path: string): string {
  return path.endsWith('/') && path !== '/' ? path.slice(0, -1) : path
}
```
